drv/stm32h7-qspi/src/lib.rs

```diff
@@ -168,8 +200,26 @@ impl Qspi {
         // off the front.
         let mut data = data;
         while !data.is_empty() {
+            let sr = self.reg.sr.read();
+
+            if sr.tof().bit_is_set() {
+                self.reg.fcr.modify(|_, w| w.ctof().set_bit());
+                self.disable_all_interrupts();
+                return Err(QspiError::Timeout);
+            }
+            if sr.tef().bit_is_set() {
+                self.reg.fcr.modify(|_, w| w.ctef().set_bit());
+                self.disable_all_interrupts();
+                return Err(QspiError::TransferError);
+            }
```

Comment:
It occurs to me that, rather than having all the error paths have to remember to explicitly disable IRQs, we could maybe wrap this in an inner function that we call and then match the `Result` from, so that there's one place where we clear the IRQs regardless of how we exited from the function. Not sure if this is worth the effort unless there's a possibility of adding additional error paths here in the future.